### ISSUE TITLE
doubt of loss calculation

### ISSUE THREAD

**Issue by OswaldoBornemann** [MilkKnight/RNN_Language_Model#1]
hello, may i ask a question about the way you calculate loss ? In the code, you write like below:

```
# To compute the logits
        logits = tf.map_fn(output_embedding, outputs)
        logits = tf.reshape(logits, [-1, vocab_size])
        loss = tf.nn.sparse_softmax_cross_entropy_with_logits(labels=tf.reshape(self.output_batch, [-1]), logits=logits) \
               * tf.cast(tf.reshape(non_zero_weights, [-1]), tf.float32)
```
i have some problem in understanding why the part of `sparse softmax cross entropy with logits` needs to multiply with `tf.cast` part ?
